sdk/core/azure-core/azure/core/tracing/common.py

```diff
@@ -0,0 +1,98 @@
+"""Common functions shared by both the sync and the async decorators."""
+
+from azure.core.tracing.context import tracing_context
+from azure.core.tracing.abstract_span import AbstractSpan
+from azure.core.settings import settings
+
+
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    from typing import Any, Optional
+
+
+def set_span_contexts(wrapped_span, span_instance=None):
+    # type: (AbstractSpan, Optional[AbstractSpan]) -> None
+    """
+    Set the sdk context and the implementation context. `span_instance` will be used to set the implementation context
+    if passed in else will use `wrapped_span.span_instance`.
+
+    :param wrapped_span: The `AbstractSpan` to set as the sdk context
+    :type wrapped_span: `azure.core.tracing.abstract_span.AbstractSpan`
+    :param span_instance: The span to set as the current span for the implementation context
+    """
+    tracing_context.current_span.set(wrapped_span)
+    impl_wrapper = settings.tracing_implementation()
+    if wrapped_span is not None:
+        span_instance = wrapped_span.span_instance
+    if impl_wrapper is not None:
+        impl_wrapper.set_current_span(span_instance)
+
+
+def get_parent_and_original_contexts(kwargs):
+    # type: (Any) -> Tuple(AbstractSpan, AbstractSpan, Any)
+    """
+    Returns the current span so that the function's span will be its child. It will create a new span if there is
+    no current span in any of the context. Because it creates a new span in those cases, also returns the original
+    span as well. If it does not create a new span then the orig_wrapped_span should be the same as the parent span.
+
+    `AbstractSpan = azure.core.tracing.abstract_span.AbstractSpan`
+
+    :param kwargs: The dictionary of named variables that the user passed into the top level azure function
+    :param kwargs: dict
+    :returns: The parent_span, the original wrapped span, the original implementation span
+    :rtype: `AbstractSpan`, `AbstractSpan`, implementation span
+    """
+    parent_span = kwargs.pop("parent_span", None)  # type: AbstractSpan
+    orig_wrapped_span = tracing_context.current_span.get()
+
+    wrapper_class = settings.tracing_implementation()
+    if wrapper_class is None:
+        return None, orig_wrapped_span, None
+    original_wrapper_span_instance = wrapper_class.get_current_span()
+    # parent span is given, get from my context, get from the implementation context or make our own
+    parent_span = orig_wrapped_span if parent_span is None else wrapper_class(parent_span)
+    if parent_span is None:
+        current_span = wrapper_class.get_current_span()
+        parent_span = (
+            wrapper_class(span=current_span)
+            if current_span
+            else wrapper_class(name="azure-sdk-for-python-first_parent_span")
+        )
+
+    return parent_span, orig_wrapped_span, original_wrapper_span_instance
```

Comment:
So if I don't pass in `parent_span`, `parent_span` is `orig_wrapped_span`, i.e. the current span, wrapped, and the third thing is the current span again, wrapped or not I'm uncertain. So setting aside wrapping questions, the return is then (current span, current span, current span)?

Reply:
Yes so you get 
`(wrapped_current_span, wrapped_current_span, wrapped_current_span_instance)`

Reply:
ok you are correct. This is too complicated. I refactored it to simplify it down. I think you will like the new function now.